engine/script/src/script.cpp

```diff
@@ -258,13 +258,76 @@ namespace dmScript
         Unref(L, LUA_REGISTRYINDEX, context->m_ContextTableRef);
     }
 
-    lua_State* GetLuaState(HContext context) {
+    lua_State* GetLuaState(HContext context)
+    {
         if (context != 0x0) {
             return context->m_LuaState;
         }
         return 0x0;
     }
 
+    // From ldblib.c (getthread)
+    static lua_State* GetLuaThread(lua_State *L, int *arg)
+    {
+        if (lua_isthread(L, 1)) {
+            *arg = 1;
+            return lua_tothread(L, 1);
+        }
+        else {
+            *arg = 0;
+            return L;
+        }
+    }
+
+    // From https://zeux.io/2010/11/07/lua-callstack-with-c-debugger/
+    // and also
+    // https://github.com/defold/defold/blob/dev/engine/lua/src/lua/ldblib.c#L321
+    void GetLuaTraceback(lua_State* L, const char* infostring, void (*cbk)(lua_State* L, lua_Debug* entry, void* ctx), void* ctx)
+    {
+        const int LEVELS1 = 12;  // size of the first part of the stack
+        const int LEVELS2 = 10;  // size of the second part of the stack
+
+        int firstpart = 1;
+        int arg = 0;
+        lua_State* L1 = GetLuaThread(L, &arg);
+
+        lua_Debug entry;
+        int level = 0;
+        bool first = true;
+
+        if (lua_isnumber(L, arg+2)) {
+            level = (int)lua_tointeger(L, arg+2);
+            lua_pop(L, 1); // problematic?!
```

Comment:
I don't see it as much of a problem in the crash handler, but what about when using it as the "debug traceback" ? Does it mess something up?